File: 3541-find-most-frequent-vowel-and-consonant/3541-find-most-frequent-vowel-and-consonant.py
class Solution(object):
    def maxFreqSum(self, s):
        hash_tb = dict()
        for val in s:
            if hash_tb.get(val) is not None:
                hash_tb[val] += 1
            else:
                hash_tb[val] = 1
        vowels = [u'a', u'i', u'u', u'e', u'o']
        max_vowels = 0
        max_consonants = 0

        for key, val in hash_tb.items():
            if key in vowels:
                if max_vowels <= val:
                    max_vowels = val
            else:
                if max_consonants <= val:
                    max_consonants = val
        
        return max_vowels + max_consonants

File: 3541-find-most-frequent-vowel-and-consonant/3541-find-most-frequent-vowel-and-consonant.rs
use std::collections::HashMap;

impl Solution {
    pub fn max_freq_sum(s: String) -> i32 {
        let mut max_vowels = 0;
        let mut max_consonants = 0;
        let vowels = vec!['a', 'e', 'i', 'o', 'u'];
        let mut dict = HashMap::new();

        let str_arr = String::from(s);


        for v in str_arr.chars() {
            if dict.contains_key(&v) {
                dict.insert(v, dict.get(&v).unwrap() + 1);
            } else {
                dict.insert(v, 1);
            }
        }

        for (key, val) in &dict {
            if vowels.contains(&key) {
                if val > &max_vowels {
                    max_vowels = *val;
                }
            } else {
                if val > &max_consonants {
                    max_consonants = *val;
                }
            }
        }

        return max_vowels + max_consonants;
    }
}